### DISCUSSION TITLE
Documentation / Examples of edgeApi->downloadZip()

### DISCUSSION THREAD

**Discussion by tacman** [ToshY/BunnyNet-PHP#142] [Q&A] [answered]
I just noticed that you have a downloadZip() method, and was wondering how it differs from downloading a file.

I was just starting to think about how to add an option to upload/download zip files so I didn't have to manually to it.  

Is there an example of uploading or download a file or directory that uses the zip api calls?

But I guess the first question is how it differs, I've been downloading zip files using the downloadFile() call, but of course want to take advantage of whatever additional functionality is there.

Thanks again for this great library.  I've been able to replace some s3 calls, which works great but the security setup is (rightfully) complicated.  Where with bunny cdn, a single API key and a few calls via this library, it's clean and easy.

**Comment by ToshY** [comment, marked as answer]
Hey @tacman 👋 

> But I guess the first question is how it differs, I've been downloading zip files using the downloadFile() call

Could you show me an example of how you are able to download ZIP files using the `downloadFile` call? From the [API specifications](https://docs.bunny.net/reference/get_-storagezonename-path-filename) it requires a `storageZoneName`, `path` and `fileName` so I'm curious how your able to download a ZIP file currently. 

I've noticed when providing a path as `fileName` it returns the same result as `listFiles` (a JSON array with files of the specified directory) but not a ZIP, so let me know how you'd managed to achieve that. 

If it is indeed already possible to download zip with just the `downloadFile` call, I dont think there's much (or any difference) to the `downloadZip` call (and possibly can remove this method). The documentation needs updating if that's the case.

> Is there an example of uploading or download a file or directory that uses the zip api calls?

You can check this in the [documentation website](https://toshy.github.io/BunnyNet-PHP/edge-storage-api/#download-zip).

**Reply by tacman** [reply]
Ah, I see.  downloadZip() takes a _path_ in the zone and downloads all the files from that path as a zip file.  

I am looking for the opposite, to zip files in a directory and upload them (as a zip file), then when I call downloadZip, it downloads the zip file and unzips them.  So indeed, from my application, I am just downing a zip file, and my application can do the zipping and unzipping (and continue using the file calls).

BUT now that I know I can download a while directory as a zip file, I'll do that.  

Thanks.
